### PR TITLE
Add a Rect.largest constructor equivalent to SkRect::MakeLargest

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

**Comment by jason-simmons** [comment]
@Hixie

**Comment by cbracken** [comment]
lgtm

File: lib/ui/geometry.dart [hunk 1]
```diff
@@ -596,6 +596,16 @@ class Rect {
       ..[3] = math.max(a.dy, b.dy);
   }
 
+  /// Construct the largest finite rectangle.
+  Rect.largest() {
+    const double skScalarMax = 3.402823466e+38;
```
Comment (chinmaygarde) [inline lib/ui/geometry.dart]: `std::numeric_limits::max()`

Comment (jason-simmons) [reply lib/ui/geometry.dart]: afaik Dart doesn't have an API like that for single precision floats

Comment (Hixie) [inline lib/ui/geometry.dart]: document where this number comes from, in case someone has to debug this code one day

**Comment by Hixie** [comment]
LGTM

**Pull request merged**